### PR TITLE
Fix query is always an array

### PR DESCRIPTION
**Pull request by adiux**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging or 3.1
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          |no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | no
| Related developer documentation PR URL | no
| Issue(s) addressed                     | 

#### Description:
Fixes 500 server error when parameters are attached to an url 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a landingage. e.g. test
3. Hit it without being logged in (or tracking won't work). E.g. https://mautibox.com/staging/test?utm_source=linkedin&utm_medium=social&utm_campaign=mautic&utm_content=testing

### CODE REVIEW

**Comment by escopecz** [comment]
@adiwegs how is this PR different from https://github.com/mautic/mautic/pull/9265?

File: app/bundles/PageBundle/Model/PageModel.php [hunk 1]
```diff
@@ -765,10 +765,8 @@ public function processPageHit(Hit $hit, $page, Request $request, Lead $lead, $t
 
     /**
      * @param Redirect|Page|null $page
-     *
-     * @return array
      */
-    public function getHitQuery(Request $request, $page = null)
+    public function getHitQuery(Request $request, $page = null): array
```
Comment (escopecz) [inline app/bundles/PageBundle/Model/PageModel.php]: Changing signature of a public of protected method is considered a BC break.

**Review by escopecz — CHANGES_REQUESTED** [review]

**Comment by adiux** [comment]
This is a bug fix that is also included in #9265. I separated it to test the fix without the rest of the code.

**Comment by escopecz** [comment]
The code looks great. Now could you write a functional test for this? It would ensure that the URL with query params works and will continue to work after future changes. See for example https://github.com/mautic/mautic/blob/51726b01aa718083649d84228a80e5023e0d58cc/app/bundles/LeadBundle/Tests/Controller/CompanyControllerTest.php#L43

A page can be created in the test like
```php
$page = new Page();
$page->set... // Set all the required properties
$this->em->persist($page);
$this->em->flush();

// And then access the URL for example like
$this->client->request(Request::GET, "{$page->getAlias()}?utm_source=linkedin&utm_medium=social&utm_campaign=mautic&utm_content=testing");
$clientResponse = $this->client->getResponse();
$this->assertEquals(Response::HTTP_OK, $clientResponse->getStatusCode());
```

**Comment by adiux** [comment]
I have added the Tests from #9265 to this branch too. So this fix has tests and can be merged.

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by kuzmany** [comment]
@adiux Does it prevent the issue?

`PHP Warning - Invalid argument supplied for foreach() - in file /pathtomautic/app/bundles/PageBundle/Model/PageModel.php - at line 1179`

**Comment by adiux** [comment]
@kuzmany yes

**Comment by adiux** [comment]
Closed in favour of https://github.com/mautic/mautic/pull/9265

**Pull request closed without merging**